Sources/Extensions/AppIntents/Widget/Gauge/WidgetGaugeAppIntentTimelineProvider.swift

```diff
@@ -97,19 +98,30 @@ struct WidgetGaugeAppIntentTimelineProvider: AppIntentTimelineProvider {
         }
 
         let params = renderedTemplate?.split(separator: "|") ?? []
-        guard params.count == 4 else {
+        guard params.count == 5 else {
             Current.Log.error("Failed to render template for gauge widget: Wrong length response")
             throw WidgetGaugeDataError.badResponse
         }
 
         let valueText = String(params[1])
         let maxText = String(params[2])
         let minText = String(params[3])
+        let labelText = String(params[4])
 
         let action = await withCheckedContinuation { continuation in
+            var hasResumed = false
+
             configuration.action?.asAction { action in
+                guard !hasResumed else { return }
+                hasResumed = true
                 continuation.resume(returning: action)
             }
+
+            // Fallback: Ensure continuation is resumed in case `asAction` doesn't call the closure in time
```

Comment:
I think this extra logic does not make sense given the `withCheckedContinuation` `await` already waits for the `asAction`.
What you could do is checking it configuration.action is not nil before moving forward

Reply:
I have fixed this in a separate PR, after I merge it, feel free to rebase:
https://github.com/home-assistant/iOS/pull/2942

Reply:
rebased